packages/pluggableWidgets/datagrid-web/src/utils/__tests__/useCellRenderer.spec.ts

```diff
@@ -0,0 +1,113 @@
+import { GUID } from "mendix";
+import { renderHook } from "@testing-library/react-hooks";
+import { useCellRenderer } from "../useCellRenderer";
+import {
+    dynamicValue,
+    buildListExpression,
+    buildWidgetValue,
+    EditableValueBuilder,
+    ListAttributeValueBuilder
+} from "@mendix/piw-utils-internal";
+import { ColumnsType } from "../../../typings/DatagridProps";
+
+const col: ColumnsType = {
+    showContentAs: "attribute",
+    header: dynamicValue("Test"),
+    sortable: false,
+    resizable: false,
+    draggable: false,
+    hidable: "no" as const,
+    width: "autoFill" as const,
+    size: 1,
+    alignment: "left" as const,
+    wrapText: false
+};
+
+describe("Datagrid useCellRenderer hook", () => {
+    it("adds title when column has attribute as datasource", () => {
+        const tooltipValue = "Jupiter is the fifth planet from the Sun";
+        const { result } = renderHook(() => {
+            const attribute = new ListAttributeValueBuilder().withId("attribute1").withType("String").build();
+            const tooltip = buildListExpression(tooltipValue);
+
+            attribute.get = jest.fn(() => new EditableValueBuilder<string>().withValue("Jupiter").build());
+
+            return useCellRenderer({
+                columns: [{ ...col, attribute, tooltip }]
+            });
+        });
+
+        const renderWrapper = jest.fn(x => x);
+        const output = result.current(renderWrapper, { id: "111" as GUID }, 0);
+        expect(renderWrapper).toBeCalled();
+        expect(output.type).toBe("span");
+        expect(output.props.title).toEqual(tooltipValue);
+        expect(output.props.children).toEqual("Formatted Jupiter");
```

Comment:
I am wondering why the children is totally different from the parameters..

Reply:
This is default "formatter" that we have in `ListAttributeValueBuilder`